src/Symfony/Component/ObjectMapper/Tests/Fixtures/TransformCollection/A.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Symfony\Component\ObjectMapper\Tests\Fixtures\TransformCollection;
+
+use Symfony\Component\ObjectMapper\Attribute\Map;
+use Symfony\Component\ObjectMapper\Transform\MapCollection;
+
+class A
+{
+    #[Map(transform: new MapCollection())]
+    /** @var C[] */
+    public array $foo;
```

Comment:
Question: is the `#[Map]` attribute optional here? I see you omitted it in class `B`. The `array` type combined with the `C[]` doc type carry enough information for the mapping.

Reply:
we don't read the types while mapping therefore the mapping on `C` is important (`#[Map(target: D::class)]`). 

I specified the target for A to B in the test: 

```
        $transformed = $mapper->map($u, TransformCollectionB::class);
```

Reply:
Okay, got it.